### ISSUE TITLE
Support used self-signed certificates LDAP.

### ISSUE THREAD

**Issue by veily** [dexidp/dex#1277]
For LDAP server that used self-signed certificates and  demanded TLS client verification,dex should load self-signed  certificates for client.
This is my solution.
1. add clientCA & clientKey config.
```go
	// Path to a trusted root certificate file.
	RootCA string `json:"rootCA"`
	// Path to a client cert file generated by rootCA.
	ClientCert string `json:"clientCert"`
	// Path to a client private key file generated by rootCA.
	ClientKey string `json:"clientKey"`
	// Base64 encoded PEM data containing root CAs.
	RootCAData []byte `json:"rootCAData"`
```
2. Load self-signed certificates.
```go
        if c.ClientKey != "" && c.ClientCert != "" {
                cert, err := tls.LoadX509KeyPair(c.ClientCert, c.ClientKey)
                if err != nil {
                        return nil, fmt.Errorf("ldap: load self signed certs file failed.error:%s", err)
                }
                tlsConfig.Certificates = append(tlsConfig.Certificates, cert)
        }
```
3. Setting  self-signed  certificates in config file.
```yml
connectors:
- type: ldap
  name: OpenLDAP
  id: ldap
  config:
    host: localhost:636
    rootCA: /etc/dex/sslcerts/ca.crt
    clientCA: /etc/dex/sslcerts/client.crt
    clientKey: /etc/dex/sslcerts/client.key
```

**Comment by ericchiang** [comment]
Which providers do this? Got some docs?

**Comment by veily** [comment]
Used for LDAP server that used self-signed certificates (RootCA, server cert) and setting TLS verify client. In this case, dex should use the client certificates generated by the Ldap server RootCA.
Without the client certificates, dex all requests to ldap server will be rejected.


http://www.openldap.org/doc/admin24/tls.html
> 16.1.2. Client Certificates
The DN of a client certificate can be used directly as an authentication DN.
...

**Comment by srenatus** [comment]
Fixed in v2.12.0 (#1314), thanks to your help in #1278. 🎉
